lib/temporal/connection/grpc.rb

```diff
@@ -117,9 +118,13 @@ def start_workflow_execution(
 
         client.start_workflow_execution(request)
       rescue ::GRPC::AlreadyExists => e
-        # Feel like there should be cleaner way to do this...
-        run_id = e.details[/RunId: ([a-f0-9]+-[a-f0-9]+-[a-f0-9]+-[a-f0-9]+-[a-f0-9]+)/, 1]
-        raise Temporal::WorkflowExecutionAlreadyStartedFailure.new(e.details, run_id)
+        cast_error = e.to_rpc_status&.details&.map do |any_error|
+          next unless any_error.type_url == 'type.googleapis.com/temporal.api.errordetails.v1.WorkflowExecutionAlreadyStartedFailure'
+
+          any_error.unpack(Temporal::Api::ErrorDetails::V1::WorkflowExecutionAlreadyStartedFailure)
+        end&.compact&.first
```

Comment:
I think this approach works, however there must be a way to unpack every error without checking it's type explicitly and then pick the right one. I have a sense that it might be useful in other places as well and this approach will not work if there's a change in the error type definition

Reply:
Looking at other languages, the [pattern](https://github.com/grpc/grpc-go/blob/master/examples/features/errors/client/main.go#L55-L67) seems to be to loop over and do type checks. I went ahead and moved to that form.

I also noticed there was a gap with `signal_with_start_workflow_execution` so that was updated to have this error handling